### PR TITLE
account: Adds AccountManager

### PR DESCRIPTION
**Pull request by AcK77**
In a way to have Custom User Profiles merged in master faster, this PR adds a `AccountManager` class (based on `AccountUtils` class) and the following changes have been made:
- Adds a "default profile values" which were the old hardcoded ones.
- The image profile is moved to the Account service folder.
- The hardcoded UserId for the savedata is now using the `AccountManager` last opened one.
- The DeviceId in Mii service is changed to the right value (checked by REd sys:set call).
- Some cleanups.

### CODE REVIEW

File: Ryujinx.HLE/HOS/Services/Account/Acc/AccountManager.cs [hunk 1]
```diff
@@ -1,23 +1,32 @@
...
-        internal UserProfile LastOpenedUser { get; private set; }
+        public UserProfile LastOpenedUser { get; private set; }
 
-        public AccountUtils()
+        public AccountManager()
         {
             _profiles = new ConcurrentDictionary<string, UserProfile>();
+
+            UserId defaultUserId          = new UserId("00000000000000010000000000000000");
+            Stream defaultUserImageStream = GetType().GetTypeInfo().Assembly.GetManifestResourceStream("Ryujinx.HLE.HOS.Services.Account.Acc.DefaultUserImage.jpg");
```
Comment (gdkchan) [inline Ryujinx.HLE/HOS/Services/Account/Acc/AccountManager.cs:20]: Maybe you can use https://github.com/Ryujinx/Ryujinx/blob/master/Ryujinx.Common/Utilities/EmbeddedResources.cs#L113 ?

File: Ryujinx.HLE/HOS/Services/Account/Acc/Types/UserProfile.cs [hunk 2]
```diff
@@ -1,9 +1,9 @@
-﻿using Ryujinx.HLE.Utilities;
-using System;
+﻿using System;
+using System.IO;
 
 namespace Ryujinx.HLE.HOS.Services.Account.Acc
 {
-    class UserProfile
+    public class UserProfile
```
Comment (gdkchan) [inline Ryujinx.HLE/HOS/Services/Account/Acc/Types/UserProfile.cs:6]: If this needs to be public, I would move it to `Ryujinx.HLE/HOS`.

File: Ryujinx.HLE/HOS/Services/Account/Acc/Types/UserProfile.cs [hunk 3]
```diff
@@ -13,14 +13,18 @@ class UserProfile
 
         public long LastModifiedTimestamp { get; private set; }
 
+        public Stream ImageStream { get; private set; }
```
Comment (gdkchan) [inline Ryujinx.HLE/HOS/Services/Account/Acc/Types/UserProfile.cs:16]: Why not store a byte array with the image data, rather than the stream? Btw, `private set` is not needed if you set the property from the constructor, so you can remove it from this and the other properties.

File: Ryujinx.HLE/HOS/Services/Mii/Helper.cs [hunk 4]
```diff
@@ -32,7 +31,7 @@ public static ushort CalculateCrc16BE(ReadOnlySpan<byte> data, int crc = 0)
         public static UInt128 GetDeviceId()
         {
             // FIXME: call set:sys GetMiiAuthorId
-            return SystemStateMgr.DefaultUserId.ToUInt128();
+            return new UInt128(new Guid().ToByteArray());
```
Comment (gdkchan) [inline Ryujinx.HLE/HOS/Services/Mii/Helper.cs:34]: Why was this changed?

File: Ryujinx.HLE/HOS/SystemState/SystemStateMgr.cs [hunk 5]
```diff
@@ -1,11 +1,9 @@
-using Ryujinx.HLE.HOS.Services.Account.Acc;
 using System;
 
 namespace Ryujinx.HLE.HOS.SystemState
 {
     public class SystemStateMgr
     {
-        public static readonly UserId DefaultUserId = new UserId("00000000000000010000000000000000");
```
Comment (gdkchan) [inline Ryujinx.HLE/HOS/SystemState/SystemStateMgr.cs:7]: Nit: Remove empty line.

File: Ryujinx.HLE/HOS/Services/Mii/Helper.cs (hunk 4 above)
Comment (AcK77) [reply Ryujinx.HLE/HOS/Services/Mii/Helper.cs:34]: Because it was false before, it's return the DeviceId and a `nn::util::GenerateUuid()` if the DeviceId is null.

File: Ryujinx.HLE/HOS/Services/Account/Acc/Types/UserProfile.cs (hunk 2 above)
Comment (AcK77) [reply Ryujinx.HLE/HOS/Services/Account/Acc/Types/UserProfile.cs:6]: I prefer to keep this class in Account service, since it's where the struct UserProfile is stored in the real service.

File: Ryujinx.HLE/HOS/Services/Mii/Helper.cs [hunk 6]
```diff
@@ -32,7 +31,7 @@ public static ushort CalculateCrc16BE(ReadOnlySpan<byte> data, int crc = 0)
         public static UInt128 GetDeviceId()
         {
             // FIXME: call set:sys GetMiiAuthorId
-            return SystemStateMgr.DefaultUserId.ToUInt128();
+            return new UInt128(0, 1);
```
Comment (Thog) [inline Ryujinx.HLE/HOS/Services/Mii/Helper.cs:34]: Nice catch 👍

**Review by Thog — APPROVED** [review]
lgtm

**Review by riperiperi — APPROVED** [review]
LGTM 👍

File: Ryujinx/Ui/Widgets/GameTableContextMenu.cs [hunk 7]
```diff
@@ -429,7 +432,7 @@ public Result CopyFile(FileSystemClient fs, string sourcePath, string destPath)
         private void OpenSaveUserDir_Clicked(object sender, EventArgs args)
         {
             SaveDataFilter saveDataFilter = new SaveDataFilter();
-            saveDataFilter.SetUserId(new UserId(1, 0)); // TODO: Remove Hardcoded value.
+            saveDataFilter.SetUserId(new LibHac.Fs.UserId((ulong)_accountManager.LastOpenedUser.UserId.High, (ulong)_accountManager.LastOpenedUser.UserId.Low));
```
Comment (gdkchan) [inline Ryujinx/Ui/Widgets/GameTableContextMenu.cs:435]: Why not just `_accountManager.LastOpenedUser.UserId.ToLibHacUid()`.

**Review by gdkchan — APPROVED** [review]
Other than my last comment, lgtm.

File: Ryujinx/Ui/Widgets/GameTableContextMenu.cs (hunk 7 above)
Comment (AcK77) [reply Ryujinx/Ui/Widgets/GameTableContextMenu.cs:435]: Because ToLibHacUid() != LibHac.Fs.UserId
